### ISSUE TITLE
Package request: gnome-remote-desktop

### ISSUE THREAD

**Issue by hksdpc250** [clearlinux/distribution#2370]
Official package name: gnome-remote-desktop

License (must be an OSI approved Open Source license): GNU GPLv2

Download URL of latest release: https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/archive/fcff89c168a35471a014453f15b13c78eba076a5/gnome-remote-desktop-fcff89c168a35471a014453f15b13c78eba076a5.tar.bz2

Latest release date (must be recent): May 1, 2021 18:25 GMT+0000

Description: Remote desktop daemon for GNOME using pipewire.

**Comment by hksdpc250** [comment]
Currently, this package is the only solution to enable remote desktop feature for GNOME on Wayland.

**Comment by Aqua1ung** [comment]
Yeah, at this point in time CL doesn't have any Wayland-based, non-VNC based remote desktop solution.

**Comment by fenrus75** [comment]
ok finally managed to tame this guy; it has very nasty cuda dependencies
and the like but it should be available soon

**Comment by Aqua1ung** [comment]
> ok finally managed to tame this guy; it has very nasty cuda dependencies and the like but it should be available soon

Many thanks! I'll give it a try and report back. (Currently running CL 38700.) Otherwise, how about trying your hand at [NoMachine](https://www.nomachine.com/) :-) , now, that you're on fire? As a matter of fact, the work for NoMachine has been [90% done by this guy](https://community.clearlinux.org/t/nomachine-anyone/8487/11?u=aqua1ung), so the rest of it should be a mere formality, such as sorting out license issues and packaging stuff.

Anyway, if gnome-remote-desktop works (which I have yet to test), NoMachine is less of a pressing issue afaiac. [This one,](https://github.com/clearlinux/distribution/issues/2844) on the other hand, might score you some points with the "mere clicker" crowds, and help make CL more friendly, and ensure a smoother adoption. Just something to think about.

Now, regardless of the outcome of these (arguably inopportune) requests, please receive my gratitude for finally dealing w/gnome-remote-desktop!

**Comment by fenrus75** [comment]
NoMachine does not appear to be open source -- no go

**Comment by Aqua1ung** [comment]
[Input Remapper](https://github.com/sezanzeb/input-remapper/releases/tag/2.0.0) _is_ open source though :-)

Anyway, so far I've had trouble using gnome-remote-desktop: Remmina crashes after connecting, and Gnome Connections looks like it connects (outputs a "Success" message), but doesn't show anything. Will keep trying, and will keep you posted.

Finally, I have been [struggling to make xrdp work](https://github.com/neutrinolabs/xrdp/issues/2599) for the past few weeks--it's in the [os-utils-gui bundle](https://clearlinux.org/software/bundle/os-utils-gui)--to no avail, so it looks like that one might use some massaging as well, if you happen to have nothing else better to do, that is. :-)

**Comment by Aqua1ung** [comment]
@fenrus75: Well, it's "official," after wrestling with gnome-remote-desktop for a while, I just could not make it work. Quite possibly missing some dependency or other. Just curious, have you received any reports about people _successfully_ using g-r-d?

**Comment by Aqua1ung** [comment]
@fenrus75: Got some [promising results](https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/issues/149#note_1733782) attempting RDP connections via gnome-remote-desktop. Unfortunately, connection does not fully materialize due to what appears to be client authentication failure, even though I am pretty sure the credentials are correct; here's what `journalctl -b` says:
```
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:464] [2788:2952] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:564] [2788:2952] [WARN][com.winpr.negotiate] - AcceptSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:565] [2788:2952] [ERROR][com.winpr.sspi.NTLM] - Message Integrity Check (MIC) verification failed!
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:565] [2788:2952] [WARN][com.winpr.sspi] - CompleteAuthToken status SEC_E_MESSAGE_ALTERED [0x8009030F]
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:565] [2788:2952] [WARN][com.freerdp.core.nla] - CompleteAuthToken status SEC_E_MESSAGE_ALTERED [0x8009030F]
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:565] [2788:2952] [ERROR][com.freerdp.core.transport] - client authentication failure
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:565] [2788:2952] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_INITIAL - rdp_server_accept_nego() fail
Apr 28 13:45:17 DadsGram gnome-remote-desktop-daemon[2788]: [13:45:17:565] [2788:2952] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
Apr 28 13:45:17 DadsGram gnome-remote-de[2788]: Unable to check file descriptor, closing connection
```
What am I doing wrong?

**Comment by fenrus75** [comment]
the other side does not require RC4, but only modern crypto I hope ?

**Comment by Aqua1ung** [comment]
> the other side does not require RC4, but only modern crypto I hope ?
I really wouldn't know. Just FYI, connecting to gnome-remote-desktop server works just fine in Ubuntu.

**Comment by fenrus75** [comment]
our openssl has RC4 crypto disabled so we had to patch out some hardcoded
RC4 use in the RDP stack.
(we'd have to do quite a bit of paperwork to justify a security exception..
and not sure we'd get it. RC4 was not exactly secure in 2003, never mind
2023)

**Comment by Aqua1ung** [comment]
I don't think any of the RDP clients I use requires RC4 encryption, hence I don't think the "credentials refused" error is due to the client.

**Comment by Aqua1ung** [comment]
Happy to report that gnome-remote-desktop works fine in the latest CL releases. (Tested on 39750.)
